pipeline/publish_site.py

```
@@ -0,0 +1,793 @@
+def known_keys(collection: dict) -> tuple[set[str], set[str]]:
+    """Все слаги и нормализованные ссылки файла — без окна «последние N»."""
+    slugs = {item["slug"] for item in collection["items"]}
+    urls = {normalize_url(item["source_url"]) for item in collection["items"]}
+    return slugs, urls
+
+
+# --- живость первоисточника ---------------------------------------------------
+
+
+def _http_status(url: str, timeout: int = HTTP_TIMEOUT) -> int:
+    request = urllib.request.Request(url, headers={"User-Agent": USER_AGENT})
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)